src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -961,12 +962,60 @@ private void DecodeInterceptsLocationAttribute(DecodeWellKnownAttributeArguments
             const int lineNumberParameterIndex = 1;
             const int characterNumberParameterIndex = 2;
 
-            if (!attributeSyntax.SyntaxTree.Options.Features.ContainsKey("InterceptorsPreview"))
+            if (!attributeSyntax.SyntaxTree.Options.Features.ContainsKey("InterceptorsPreview")
+                && !(attributeSyntax.SyntaxTree.Options.Features.TryGetValue("InterceptorsPreviewNamespaces", out var rawInterceptorsNamespaces) && rawInterceptorsNamespaces.Length != 0))
             {
-                diagnostics.Add(ErrorCode.ERR_InterceptorsFeatureNotEnabled, attributeSyntax);
+                // InterceptorsPreview feature flag wasn't specified, and a non-empty value for InterceptorsPreviewNamespaces wasn't specified.
+                var namespaceNames = getNamespaceNames();
+                reportFeatureNotEnabled(diagnostics, attributeSyntax, namespaceNames);
+                namespaceNames.Free();
                 return;
             }
 
+            if (attributeSyntax.SyntaxTree.Options is CSharpParseOptions { InterceptorsPreviewNamespaces: { IsEmpty: false } interceptorNamespaces })
+            {
+                // when InterceptorsPreviewNamespaces are present, ensure the interceptor is within one of the indicated namespaces
```

Comment:
>ensure the interceptor is within one of the indicated namespaces [](http://example.com/codeflow?start=67&length=64)

Consider extracting this to a helper method: `static bool IsContainedInNamespace(this Symbol symbol, ImmutableArray<string> namespaceNames)`